src/bindings/node/node.cc

```diff
@@ -0,0 +1,196 @@
+      size_t argc_ = 1;
+      napi_value args[1];
+      status = napi_get_cb_info(env, info, &argc_, args, nullptr, nullptr);
+      checkNapiStatus(status, env, "Failed to parse input args");
+
+      const size_t argc = 1;
+      napi_value argv[argc] = {args[0]};
+
+      napi_value actor_constructor;
+      status = napi_get_reference_value(env, constructor, &actor_constructor);
+      checkNapiStatus(status, env, "Failed to reference the constructor");
+
+      napi_value instance;
+      status = napi_new_instance(env, actor_constructor, 0, NULL, &instance);
+      checkNapiStatus(status, env, "Failed to create new instance");
+
+      return instance;
+    }
+  }
+
+  static std::string ParseRequest(napi_env env, napi_callback_info info, napi_value* jsthis) {
+    napi_status status;
+    size_t argc = 1;
+    napi_value argv[1];
+
+    status = napi_get_cb_info(env, info, &argc, argv, jsthis, nullptr);
+    checkNapiStatus(status, env, "Failed to parse input args");
+
+    size_t request_str_size;
+    status = napi_get_value_string_utf8(env, argv[0], nullptr, 0, &request_str_size);
+    checkNapiStatus(status, env, "Failed to get arg string length");
+
+    if (request_str_size > 1024 * 1024) {
+      napi_throw_error(env, NULL, "Too large JSON config");
+    }
+
+    char request_string[++request_str_size];
+    status =
+        napi_get_value_string_utf8(env, argv[0], request_string, request_str_size, &request_str_size);
+    checkNapiStatus(status, env, "Unable to parse arg string");
+    std::string reqString(request_string, request_str_size);
+
+    return reqString;
+  }
+
+  static napi_value WrapString(napi_env env, std::string cppStr) {
+    napi_value napiStr;
+    napi_status status;
+    const char* outBuff = cppStr.c_str();
+    const auto nchars = cppStr.size();
+    status = napi_create_string_utf8(env, outBuff, nchars, &napiStr);
+    checkNapiStatus(status, env, "Failed to turn c++ string into napi_value");
+    return napiStr;
+  }
+
+  static napi_value Route(napi_env env, napi_callback_info info) {
+    napi_value jsthis;
+    napi_status status;
+    // parse input arg into string
+    std::string reqString = ParseRequest(env, info, &jsthis);
+
+    Actor* obj;
+    status = napi_unwrap(env, jsthis, reinterpret_cast<void**>(&obj));
+    checkNapiStatus(status, env, "Failed to unwrap js object");
+
+    // get the actual route
+    std::string route_json;
+    try {
+      route_json = obj->actor.route(reqString);
+    } catch (const std::exception& e) { napi_throw_error(env, NULL, e.what()); }
+
+    // wrap route_json in napi value for return
+    auto outStr = WrapString(env, route_json);
+    return outStr;
+  }
+
+  static napi_ref constructor;
+
+  valhalla::tyr::actor_t actor;
+  napi_env env_;
+  napi_ref wrapper_;
+};
+
+napi_ref Actor::constructor;
+
+napi_value Init(napi_env env, napi_value exports) {
+  return Actor::Init(env, exports);
```

Comment:
can we add config as an argument here and configure the logging from it, something like https://github.com/valhalla/valhalla/blob/de5c2e162da568be8613ba4b1284e5cc22fa0f86/src/bindings/python/python.cc#L23-L51

Reply:
we parse the config in the `New` function - is that where we would configure the logging?

Reply:
oh okay I see - so I should configure the logging when we init, and then even if we re-set the config, the logging will remain configured?

Reply:
yep exactly. its only a small lie because the logging config and the config for other stuff are in the same file. if we had logging config separate from the rest it wouldnt be a problem. basically we need a little bit of a refactor to clean this up. but for now, yep, exactly what you described will work!